File: python/src/provider.py
class Provider:
    def __init__(self, client, context, profile):
        self.client = client
        self.context = context
        self.profile = profile

    def put(self, property, value):
        system_id = self.context.node.system.id
        node_id = self.context.node.id
        context_id = self.context.id
        profile = self.profile
        key = f'cns/{system_id}/nodes/{node_id}/contexts/{context_id}/provider/{profile}/properties/{property}'
        self.client.put(key, value)

File: nodejs/provider.js
// Copyright 2025 Padi, Inc. All Rights Reserved.

export class Provider {
  #client;
  context;
  profile;

  constructor(client, context, profile) {
    this.#client = client;
    this.context = context;
    this.profile = profile;
  }

  put(property, value) {
    const key = `cns/${this.context.node.system.id}/nodes/${this.context.node.id}/contexts/${this.context.id}/provider/${this.profile}/properties/${property}`;
    return this.#client.put(key, value);
  }
}
